pkg/controllers/osc/resources/operating_system_config.go

```diff
@@ -72,47 +71,48 @@ func OperatingSystemConfigCreator(
 				return nil, fmt.Errorf("failed to get cloud provider from machine deployment: %v", err)
 			}
 
-			CACert, err := resources.GetCACert(kubeconfig)
+			cloudConfig, err := GetCloudConfig(cloudConfigSecret)
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("failed to get cloud config data from cloud config secret: %v", err)
 			}
-			kubeletConfig, err := kubeletConfiguration("cluster.local", clusterDNSIPs, map[string]bool{"RotateKubeletServerCertificate": true})
+
+			CACert, err := resources.GetCACert(kubeconfig)
 			if err != nil {
 				return nil, err
 			}
+
 			kubeconfigStr, err := resources.StringifyKubeconfig(kubeconfig)
 			if err != nil {
 				return nil, err
 			}
 
-			kubeletSystemdUnit, err := KubeletSystemdUnit("docker", md.Spec.Template.Spec.Versions.Kubelet, cloudProvider.Name, "node-1", clusterDNSIPs, false,
-				"", nil, KubeletFlags())
-			if err != nil {
-				return nil, err
-			}
-
-			safeDownloadBinariesScript, err := SafeDownloadBinariesScript(md.Spec.Template.Spec.Versions.Kubelet)
-			if err != nil {
-				return nil, err
+			// ensure that Kubelet version is prefixed by "v"
+			kubeVersion := md.Spec.Template.Spec.Versions.Kubelet
```

Comment:
Please use `github.com/Masterminds/semver/v3` to parse and validate Kubelet version.
```suggestion
	kubeletVersion, err := semver.NewVersion(md.Spec.Template.Spec.Versions.Kubelet)
	if err != nil {
		return "", fmt.Errorf("invalid kubelet version: %w", err)
	}

```